src/openforms/registrations/contrib/zgw_apis/options.py

```diff
@@ -104,6 +105,14 @@ class ZaakOptionsSerializer(JsonSchemaSerializerMixin, serializers.Serializer):
     )
 
     # Objects API
+    objects_api_group = PrimaryKeyRelatedAsChoicesField(
+        queryset=ObjectsAPIGroupConfig.objects.exclude(
+            Q(objects_service=None) | Q(objecttypes_service=None)
+        ),
+        help_text=_("Which Objects API set to use."),
+        label=_("Objects API set"),
+        required=False,
```

Comment:
I think we may need an `allow_null=True` to make it explicit that no objects API integration is done.

There's some more steps involved in this PR I think:

* A data migration that updates existing registration backends
    - if no `objecttype` is specified, ensure that `objects_api_group` is set to `None`
    - if an objecttype is specified, set the `objects_api_group` to the lowest PK (use that queryset that is now used at runtime)
    - this behaviour also needs to be applied during form import time when legacy forms are being imported (!), we can probably do something with the [`DEFINITION_CONVERTERS`](https://github.com/open-formulieren/open-forms/blob/ad7ead2aa3267c23ecda3063cc570992ab91ca47/src/openforms/formio/migration_converters.py#L295)
* Validation! Once we know which API group is used, we know the API root URl of the objecttypes API, and can check the URL provided. We can even look up the provided object type and check that it exists, but let's not do _that_ yet because credentials to the object types API may not be configured yet. This should only come up when people edit the form again, and should not affect existing runtime behaviour of forms.

Reply:
I added a new type of converter for the registration backends, that runs before the data is passed into the serializer